### PR TITLE
feat: update route metadata service to read from router module

### PR DESCRIPTION
**Pull request by johnhwhite**

### CODE REVIEW

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
I see provider errors when I serve the playground app.

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Paul mentioned a few days ago that we should simply copy/paste the implementation from our docs site and modify that as needed (here's a [rough copy/paste](https://github.com/blackbaud/stache/pull/28) I did to make sure it was a feasible option, and it seemed to work as a first pass).

What improvements does the `RouterPreloader` bring that aren't represented by the docs implementation?

File: apps/playground/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -57,10 +57,17 @@ const routes: Routes = [
...
+    path: 'navigation',
+    loadChildren: () =>
+      import('./navigation/navigation.module').then((m) => m.NavigationModule),
+  },
 ];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [
+    RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules }),
```
[the remark this answers is not in the document]
Comment (johnhwhite) [reply apps/playground/src/app/app-routing.module.ts]: The preloading was intended to load metadata from asynchronous routes. I've pulled it and we can revisit it later.

File: apps/playground/src/app/navigation/side-nav-routes.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { StacheNavLink } from '@blackbaud/skyux-lib-stache';
+
+export const sideNavRoutes: StacheNavLink[] = [
```
[the remark this answers is not in the document]
Comment (johnhwhite) [reply apps/playground/src/app/navigation/side-nav-routes.ts]: I'd added this to confirm how it ought to work. I've now removed it.

File: apps/playground/src/app/navigation/navigation.component.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Component } from '@angular/core';
+
+import { sideNavRoutes } from './side-nav-routes';
+
+@Component({
+  selector: 'app-navigation',
+  templateUrl: './navigation.component.html',
+  preserveWhitespaces: true,
```
[the remark this answers is not in the document]
Comment (johnhwhite) [reply apps/playground/src/app/navigation/navigation.component.ts]: This was just example content. I've replaced `preserveWhitespaces: true`.

File: apps/playground/src/app/navigation/supporting-pages/recursion/recursion.component.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Component } from '@angular/core';
+
+import { sideNavRoutes } from '../../side-nav-routes';
+
+@Component({
+  selector: 'app-recursion',
+  templateUrl: './recursion.component.html',
+  preserveWhitespaces: true,
```
[the remark this answers is not in the document]
Comment (johnhwhite) [reply apps/playground/src/app/navigation/supporting-pages/recursion/recursion.component.ts]: This was just example content. I've replaced `preserveWhitespaces: true`.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**